metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -923,8 +924,31 @@ def visit_combine_metrics_node(self, node: CombineMetricsNode[SqlDataSetT]) -> S
         OUTER JOIN context, because many engines do not support complex ON conditions or other techniques we might
         use to apply a sentinel value for NULL to NULL comparisons.
         """
-        # Sanity check that all parents have the same linkable specs.
-        parent_data_sets = [x.accept(self) for x in node.parent_nodes]
+        parent_data_sets: List[SqlDataSet] = [parent_node.accept(self) for parent_node in node.parent_nodes]
+        uses_time_offset = False
+        input_metric_specs: Dict[str, Sequence[MetricSpec]] = defaultdict(list)
+        parent_aliases_to_data_sets: OrderedDict[str, SqlDataSet] = OrderedDict()
+        table_alias_to_metric_specs: OrderedDict[str, Sequence[MetricSpec]] = OrderedDict()
+        for i, parent_node in enumerate(node.parent_nodes):
+            parent_data_set = parent_data_sets[i]
+            table_alias = self._next_unique_table_alias()
+            parent_data_sets.append(parent_data_set)
+
+            # If metric is derived, get input metrics.
+            if isinstance(parent_node, ComputeMetricsNode):
```

Comment:
You'll want to rely on the derived-ness directly, rather than node type checks, since derived metrics can be derived from other derived metrics, which might themselves be represented here as a CombineMetricsNode. Like a ratio of the sums of multiple other metrics.